.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+name: release
+
+jobs:
+  build:
+    name: Build distributions for PyPI
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Set up Python
+        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
+
+      - name: Build distributions
+        run: python setup.py sdist bdist_wheel
```

Comment:
NOTE: I copied this directly from the Makefile, but I can't remember off the top of my head whether `bdist_wheel` requires `python -m pip install wheel` first. I didn't see it anywhere else in the CI, so I _think_ it's fine?

Reply:
We don't build or publish in CI, so that doesn't prove anything ;-)

I believe installing wheel manually is required. Is there a reason not to use `pip install build` for this?

Reply:
None that I can think of, will add.

Reply:
Added; these are currently unpinned but perhaps it makes sense to hash-pin them in a `publish-requirements.txt` instead.

Reply:
Should we be using https://pypa-build.readthedocs.io/en/stable/ rather than invoking setup.py directly?

Reply:
Yep, that'd be ideal. Want me to roll that in here or do a separate PR?

Reply:
Include please

Reply:
Done; I took the liberty of also removing the `setup.py publish` patch in a separate commit (since we should never go through `setup.py` directly again), but I can revert that if you'd prefer.